docs/proposals/caching/README.md

```diff
@@ -47,82 +47,76 @@ Goals:
 
 ### Define the scope of the cached resource
 
-#### New ClusterCache API
+#### New Search APIGroup
 
-We propose a new CR in `cluster.karmada.io` group.
+We propose a new commponent named `karmada-search`, it provides a new api group called `search.karmada.io`, the reason why select the `search` word is because it's more to [OpenSearch](https://opensearch.org) which is a community-driven, Apache 2.0-licensed open source search and analytics suite.
+
+The `karmada-search` component currently supports two types of backend stores, namely `cache` and `opensearch`. It uses the `cache` type as the default backend store of the caching layer for Karamda.
+
+We introduce a new resource type called `ResourceRegistry` in `search.karmada.io` group, it requires end users to manually specify the clusters and resources that need to be cached
 
-```golang
 
+```golang
 package v1alpha1
 
 import (
-	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	clusterv1alpha1 "github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
 )
 
+const (
+	// ResourceKindResourceRegistry is the name of the resource registry
+	ResourceKindResourceRegistry = "ResourceRegistry"
+	// ResourceSingularResourceRegistry is singular name of ResourceRegistry.
+	ResourceSingularResourceRegistry = "resourceregistry"
+	// ResourcePluralResourceRegistry is plural name of ResourceRegistry.
+	ResourcePluralResourceRegistry = "resourceregistries"
+	// ResourceNamespaceScopedResourceRegistry is the scope of the ResourceRegistry
+	ResourceNamespaceScopedResourceRegistry = false
+)
```

Comment:
Maybe we don't need to add the implemented details in the proposal.

Reply:
@RainbowMango updated